src/lib.rs

```diff
@@ -227,41 +229,89 @@ pub mod pallet {
+				AccountsPayable::<T>::get(&payee).ok_or(Error::<T>::NoAssociatedClaim)?;
 			ensure!(
 				info.claimed_reward < info.total_reward,
 				Error::<T>::RewardsAlreadyClaimed
 			);
 
-			Self::make_vested_payment(info, payee.clone(), true)
-		}
+			// Vesting is done in relation with the relay chain slot
+			let now: T::BlockNumber =
+				cumulus_pallet_parachain_system::Module::<T>::validation_data()
+					.expect("validation data was set in parachain system inherent")
+					.relay_parent_number
+					.into();
 
-		/// Collect whatever portion of your reward are currently vested.
-		#[pallet::weight(0)]
-		pub fn show_me_the_money(origin: OriginFor<T>) -> DispatchResultWithPostInfo {
-			let payee = ensure_signed(origin)?;
+			// Substract the first payment from the vested amount
+			let first_paid = T::InitializationPayment::get() * info.total_reward;
 
-			// Calculate the veted amount on demand.
-			let info = AccountsPayable::<T>::get(&payee).ok_or(Error::<T>::NoAssociatedClaim)?;
-			ensure!(
-				info.claimed_reward < info.total_reward,
-				Error::<T>::RewardsAlreadyClaimed
-			);
+			// Vesting perdiods as balance
+			let vesting_period = T::VestingPeriod::get()
+				.saturated_into::<u128>()
+				.try_into()
+				.ok()
+				.ok_or(Error::<T>::WrongConversionU128ToBalance)?;
 
-			Self::make_vested_payment(info, payee.clone(), false)
-		}
+			// To calculate how much could the user have claimed already
+			let payable_period = now.saturating_sub(<InitRelayBlock<T>>::get().into());
+
+			let pay_period_as_balance: BalanceOf<T> = payable_period
+				.saturated_into::<u128>()
+				.try_into()
+				.ok()
+				.ok_or(Error::<T>::WrongConversionU128ToBalance)?;
+
+			// How much should the contributor have already claimed by this block?
+			// By multiplying first we allow the conversion to integer done with the biggest number
+			let should_have_claimed = pay_period_as_balance
+				.saturating_mul(info.total_reward - first_paid)
+				/ vesting_period;
+
+			// If the period is bigger than whats missing to pay, then return whats missing to pay
+			let payable_amount = if should_have_claimed >= info.total_reward {
+				info.total_reward.saturating_sub(info.claimed_reward)
+			} else {
+				should_have_claimed + first_paid - info.claimed_reward
+			};
+
+			// If first payment does not pay fees
+			let result: PostDispatchInfo = if !info.free_claim_done {
+				info.free_claim_done = true;
+				PostDispatchInfo {
+					actual_weight: Some(0),
+					pays_fee: Pays::No,
+				}
+			} else {
+				Default::default()
+			};
+
+			info.claimed_reward = info.claimed_reward.saturating_add(payable_amount);
+			AccountsPayable::<T>::insert(&payee, &info);
 
+			// This pallet controls an amount of funds and transfers them to each of the contributors
+			//TODO: contributors should have the balance locked for tranfers but not for democracy
+			T::RewardCurrency::transfer(
+				&PALLET_ID.into_account(),
+				&payee,
+				payable_amount,
+				KeepAlive,
```

Comment:
I believe `KeepAlive` will not allow the transfer to go through if it would take the account below the existential deposit. In this case, the ideal scenario is that every single reward token will be claimed and paid out. So if we require `KeepAlive` the last claim will not succeed because it would have killed the pallet's pot account.

Reply:
I see, I missread it then. Will change it